chart/templates/deployment.yaml

```diff
@@ -147,10 +147,11 @@ spec:
       containers:
       - name: pre-delete-controller-cleanup
         image: {{ .Values.controller.manager.image.repository }}:{{ .Values.controller.manager.image.tag }}
-        args:
-        - --finalizer-cleanup=true
         command:
         - /vault-secrets-operator
+        args:
+        - --finalizer-cleanup=true
```

Comment:
Why isn't `--finalizer-cleanup=true` set in the kustomize version?

Reply:
This job is a helm hook which gets applied to the cluster on helm delete, there isn't a real concept of that for kustomize AFAIK. 
The original reason we created the job was because helm installs were not cleaning up cleanly in ci due to us deploying default auth and connection resources via helm. 
Kustomize doesn't do this so it doesn't need to run within CI.
That being said I think this hook is getting a slight refractor in the PR that Thy is working on.